scripts/compute_product_metrics_for_all_models.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+# Iterate over results folder to get model names
+# Iterate over each input and then add a result column for each model
+
+import csv
+import json
+import os
+
+results_path = 'test-sets/results/'
+test_sets_path = 'test-sets/input/'
+
+def count_ingredients(ingredients):
+    count = 0
+    for ingredient in ingredients:
+        sub_ingredients = ingredient.get('ingredients', [])
+        if len(sub_ingredients) > 0:
+            count += count_ingredients(sub_ingredients)
+        else:
+            count += 1
+    return count
+
+        
+with open(results_path + "products_stats.csv", "w", newline="") as products_csv:
+    products_csv_writer = csv.writer(products_csv)
+
+    models = []
+
+    for model_name in [f for f in sorted(os.listdir(results_path))if '.' not in f]:
+        models.append(model_name)
+
+    products_csv_writer.writerow(['test_set','product','ingredients_n','without_ciqual_n'] + models)
+
+    for test_set_name in [f for f in sorted(os.listdir(test_sets_path))if '.' not in f]:
```

Comment:
list comprehension in for loops are barely readable, best to do:

```suggestion
    test_set_names = [f for f in sorted(os.listdir(test_sets_path))if '.' not in f]
    for test_set_name in test_set_names:
```